contracts/NFTVesting/LinearVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.5.0) (token/ERC20/utils/TokenTimelock.sol)
+// SpartanLabs Contracts (NFTVesting)
+
+pragma solidity ^0.8.0;
+import "./IERC721.sol";
+
+
+/**
+ * @dev A single NFT holder contract that will allow a beneficiary to extract the
+ * NFT after a given release time.
+ *
+ * Useful for simple vesting schedules like "whitelisted addresses get their NFT
+ * after 1 year".
+ */
+contract LinearVestingNFTTimeLock {
+
+    // ERC721 basic token smart contract
+    IERC721 private immutable _nft;
+
+    // ERC721 basic token ID of contract being held
+    uint256 private immutable _tokenId;
+
+    // beneficiary of token after they are released
+    address private immutable _beneficiary;
+
+    // timestamp when token release is enabled and when discount starts to vest
+    uint256 private immutable _releaseTime;
```

Comment:
Same re: naming, perhaps something along vesting starting?

Reply:
okay sure i think cliff might be a better naming?